addons/mrp/models/mrp_workcenter.py

```diff
@@ -195,12 +259,45 @@ def unblock(self):
     def create(self, vals_list):
         # resource_type is 'human' by default. As we are not living in
         # /r/latestagecapitalism, workcenters are 'material'
-        records = super(MrpWorkcenter, self.with_context(default_resource_type='material')).create(vals_list)
-        return records
+
+        resources_vals_list = []
+        calendar_ids = [vals['resource_calendar_id'] for vals in vals_list if vals.get('resource_calendar_id')]
+        calendars_tz = {calendar.id: calendar.tz for calendar in self.env['resource.calendar'].browse(calendar_ids)}
+        for vals in vals_list:
+            if not vals.get('resource_ids'):
+                workorder_simultaneous = vals.get('simultaneous_workorder') if vals.get('simultaneous_workorder') else 1
+                resources = []
+                for _ in range(workorder_simultaneous):
+                    resources.append(self._prepare_resource_values(
+                        vals,
+                        vals.pop('tz', False) or calendars_tz.get(vals.get('resource_calendar_id'))
+                    ))
+                resources_vals_list.append(resources)
+        if resources_vals_list:
+            resources = []
+            for r in resources_vals_list:
+                resources.append(self.env['resource.resource'].create(r))
+            resources_iter = iter(resources)
+            for vals in vals_list:
+                if not vals.get('resource_ids'):
+                    vals['resource_ids'] = next(resources_iter)
+        return super(MrpWorkcenter, self.with_context(check_idempotence=True, default_resource_type='material')).create(vals_list)
 
     def write(self, vals):
+        if vals.get('simultaneous_workorder'):
+            number_resource_to_create = vals.get('simultaneous_workorder') - len(self.resource_ids)
+            if number_resource_to_create > 0:
+                res_id = self.resource_ids[0]
+                to_create = []
+                for _ in range(number_resource_to_create):
+                    to_create.append(res_id.copy_data()[0])
+                self.resource_ids = [Command.create(a) for a in to_create]
+
+            if number_resource_to_create < 0:
```

Comment:
- `res_id` is not an ID (I suggest `first_resource` as a name but better ideas are welcome);
- I suggest a `while` loop instead of `for`, only to avoid the dummy variable, but if you want to keep the `for` loop, it's completely fine too ! (But in that case, don't use `_` as the variable's name);
- I think it's better to append the create command directll in the list, so you don't need a second loop
```suggestion
                first_resource = self.resource_ids[0]
                new_resources_vals = []
                while len(new_resources_vals) < number_resource_to_create:
                    copied_vals = first_resource.copy_data()[0]
                    new_resources_vals.append(Command.create(copied_vals))
                self.resource_ids = new_resources_vals
            elif number_resource_to_create < 0:
```